storm-core/src/jvm/org/apache/storm/executor/BaseExecutor.java

```diff
@@ -0,0 +1,153 @@
+
+    protected final ExecutorData executorData;
+    protected final Map stormConf;
+    protected final String componentId;
+    protected final WorkerTopologyContext workerTopologyContext;
+    protected final IReportError reportError;
+    protected final Callable<Boolean> sampler;
+    protected final Random rand;
+    protected final DisruptorQueue transferQueue;
+    protected final DisruptorQueue receiveQueue;
+    protected final Map<Integer, Task> idToTask;
+    protected final Map<String, String> credentials;
+    protected final Boolean isDebug;
+    protected final Boolean isEventLoggers;
+    protected String hostname;
+
+    public BaseExecutor(ExecutorData executorData, Map<Integer, Task> idToTask, Map<String, String> credentials) {
+        this.executorData = executorData;
+        this.stormConf = executorData.getStormConf();
+        this.componentId = executorData.getComponentId();
+        this.workerTopologyContext = executorData.getWorkerTopologyContext();
+        this.reportError = executorData.getReportError();
+        this.sampler = executorData.getSampler();
+        this.rand = new Random(Utils.secureRandomLong());
+        this.transferQueue = executorData.getBatchTransferWorkerQueue();
+        this.receiveQueue = executorData.getReceiveQueue();
+        this.idToTask = idToTask;
+        this.credentials = credentials;
+        this.isDebug = Utils.getBoolean(stormConf.get(Config.TOPOLOGY_DEBUG), false);
+        this.isEventLoggers = StormCommon.hasEventLoggers(stormConf);
+
+        try {
+            this.hostname = Utils.hostname(stormConf);
+        } catch (UnknownHostException ignored) {
+            this.hostname = "";
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void onEvent(Object event, long seq, boolean endOfBatch) throws Exception {
+        ArrayList<AddressedTuple> addressedTuples = (ArrayList<AddressedTuple>) event;
+        for (AddressedTuple addressedTuple : addressedTuples) {
+            TupleImpl tuple = (TupleImpl) addressedTuple.getTuple();
+            int taskId = addressedTuple.getDest();
+            if (isDebug) {
+                LOG.info("Processing received message FOR {} TUPLE: {}", taskId, tuple);
+            }
+            if (taskId != AddressedTuple.BROADCAST_DEST) {
+                tupleActionFn(taskId, tuple);
+            } else {
+                for (Integer t : executorData.getTaskIds()) {
+                    tupleActionFn(t, tuple);
+                }
+            }
+        }
+    }
+
+    public void metricsTick(Task taskData, TupleImpl tuple) {
+        try {
+            Integer interval = tuple.getInteger(0);
+            int taskId = taskData.getTaskId();
+            Map<Integer, Map<String, IMetric>> taskToMetricToRegistry = executorData.getIntervalToTaskToMetricToRegistry().get(interval);
+            Map<String, IMetric> nameToRegistry = null;
+            if (taskToMetricToRegistry != null) {
```

Comment:
there's a lot of implicit null check in clojure if we don't do the check in Java there might be an NPE.